python/util.py

```diff
@@ -0,0 +1,257 @@
+from functools import wraps
+from collections import namedtuple
+from itertools import starmap
+from operator import add, sub, eq
+
+
+def cache(method):
+    attr_name = '_' + method.__name__
```

Comment:
How does this behave with _ methods (since it will then go to __ and invoke obfuscation).